CondTools/BeamSpot/plugins/BeamProfile2DBReader.cc

```diff
@@ -0,0 +1,183 @@
+// -*- C++ -*-
+//
+// Package:    CondTools/BeamProfile2DBReader
+// Class:      BeamProfile2DBReader
+//
+/**\class BeamProfile2DBReader BeamProfile2DBReader.cc CondTools/BeamSpot/plugins/BeamProfile2DBReader.cc
+
+ Description: simple emd::one::EDAnalyzer to retrieve and ntuplize SimBeamSpot data from the conditions database
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Francesco Brivio
+//         Created:  11 June 2023
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ESWatcher.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/DataRecord/interface/SimBeamSpotObjectsRcd.h"
+#include "CondFormats/BeamSpotObjects/interface/SimBeamSpotObjects.h"
+
+// For ROOT
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include <TTree.h>
+
+#include <sstream>
+#include <fstream>
+
+//
+// class declaration
+//
+
+class BeamProfile2DBReader : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+  explicit BeamProfile2DBReader(const edm::ParameterSet&);
+  ~BeamProfile2DBReader() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+  struct theBSfromDB {
```

Comment:
`struct`-s OTOH should be uppercase, 2.6 https://cms-sw.github.io/cms_coding_rules.html

Reply:
addressed in [26deb85](https://github.com/cms-sw/cmssw/pull/41918/commits/26deb853727d29dfe01aa066a676afc14b8001a0)